bigcache.go

```diff
@@ -137,6 +137,47 @@ func (c *BigCache) Get(key string) ([]byte, error) {
 	return shard.get(key, hashedKey)
 }
 
+// Used to sore information about keys in GetMulti function
+// order is the index in the slice the data should go
+// hashedKey is the Sum64 hash of the key
+// key is the original key input
+type keyInfo struct {
+	order     int
+	hashedKey uint64
+	key       string
+}
+
+// GetMulti reads entry for each of the keys.
+// It returns an ErrEntryNotFound when
+// no entry exists for the given key.
+func (c *BigCache) GetMulti(keys []string) ([][]byte, error) {
+	shards := make(map[uint64][]keyInfo)
+	entries := make([][]byte, len(keys))
+
+	for i, key := range keys {
+		hashedKey := c.hash.Sum64(key)
+		shardIndex := hashedKey & c.shardMask
+		shards[shardIndex] = append(shards[shardIndex], keyInfo{order: i, hashedKey: hashedKey, key: key})
+	}
+
+	for shardKey, keyInfos := range shards {
+		shard := c.shards[shardKey]
+		shard.lock.RLock()
+
+		for i := range keyInfos {
+			entry, err := shard.getWithoutLock(keyInfos[i].key, keyInfos[i].hashedKey)
+
+			if err != nil {
+				shard.lock.RUnlock()
+				return nil, err
+			}
+            entries[keyInfos[i].order] = entry
```

Comment:
I think that's important to document that entries are in the same order as requested keys, and if entry is missing then it's empty. On the other handmaybe it's better to return foundEntries, and not found keys

Reply:
ok, so if a key is not found the index in the entries slice of that key should be empty. then we just keep going and returning whatever is found?

so if ["k1","k2","k3"] and k2 is empty we return ["k1val",nil,"k3val"]